ui-admin/src/study/surveys/AnswerMappingEditor.tsx

```diff
@@ -0,0 +1,303 @@
+import React, { useEffect, useMemo, useState } from 'react'
+import { faCheck, faPlus, faX } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import Select from 'react-select'
+import Creatable from 'react-select/creatable'
+import {
+  AnswerMapping,
+  AnswerMappingMapType,
+  AnswerMappingTargetType,
+  FormContent,
+  getFormElements
+} from '@juniper/ui-core'
+import { isEmpty } from 'lodash'
+import { Modal, ModalBody, ModalFooter, ModalHeader, ModalTitle } from 'react-bootstrap'
+import { OnChangeAnswerMappings } from '../../forms/formEditorTypes'
+import { ColumnDef, getCoreRowModel, useReactTable } from '@tanstack/react-table'
+import { basicTableLayout } from '../../util/tableUtils'
+import { faTrashCan } from '@fortawesome/free-solid-svg-icons/faTrashCan'
+
+
+const AnswerMappingLabels: {[key in AnswerMappingTargetType]: string} = {
+  'PROXY': 'Proxy',
+  'PROFILE': 'Profile',
+  'PROXY_PROFILE': 'Proxy Profile'
+}
+
+const ProfileFields: { [key: string]: AnswerMappingMapType } = {
+  'givenName': 'STRING_TO_STRING',
+  'familyName': 'STRING_TO_STRING',
+  'contactEmail': 'STRING_TO_STRING',
+  'phoneNumber': 'STRING_TO_STRING',
+  'birthDate': 'STRING_TO_LOCAL_DATE',
+  'doNotEmail': 'STRING_TO_BOOLEAN',
+  'doNotEmailSolicit': 'STRING_TO_BOOLEAN',
+  'sexAtBirth': 'STRING_TO_STRING',
+  'mailingAddress.street1': 'STRING_TO_STRING',
+  'mailingAddress.street2': 'STRING_TO_STRING',
+  'mailingAddress.city': 'STRING_TO_STRING',
+  'mailingAddress.state': 'STRING_TO_STRING',
+  'mailingAddress.postalCode': 'STRING_TO_STRING',
+  'mailingAddress.country': 'STRING_TO_STRING'
+}
+
+const ProxyFields: { [key: string]: AnswerMappingMapType } = {
+  'isProxy': 'STRING_TO_BOOLEAN'
+}
+
+const AnswerMappingTargets = {
+  'PROFILE': ProfileFields,
+  'PROXY': ProxyFields,
+  'PROXY_PROFILE': ProfileFields
+}
+
+const AnswerMappingMapTypeLabels: { [key in AnswerMappingMapType]: string} = {
+  'STRING_TO_STRING': 'Text',
+  'STRING_TO_BOOLEAN': 'Boolean',
+  'STRING_TO_LOCAL_DATE': 'Date'
+}
+
+type AnswerMappingRow = Partial<AnswerMapping> & { isLastRow?: boolean }
+
+/**
+ * Table which allows viewing, deleting, and creating new answer mappings.
+ */
+export default function AnswerMappingEditor(
+  {
+    formContent, initialAnswerMappings, onChange
+  } : {
+    formContent: FormContent, initialAnswerMappings: AnswerMapping[], onChange: OnChangeAnswerMappings
+  }
+) {
+  const [answerMappings, setAnswerMappings] = useState<AnswerMapping[]>(initialAnswerMappings || [])
+
+  const [openDeleteMappingModal, setOpenDeleteMappingModal] = useState<AnswerMapping | null>(null)
```

Comment:
rename so it's clearer what's being stored
```suggestion
  const [mappingSelectedForDeletion, setMappingSelectedForDeletion] = useState<AnswerMapping | null>(null)
```